### PR TITLE
Update Button to latest MUI

### LINKED ISSUE #406: Update flat button
https://material-ui.com/components/buttons/

### LINKED ISSUE #407: Update raised button
https://material-ui.com/components/buttons/

### LINKED ISSUE #409: Update icon button
https://material-ui.com/components/buttons/

### PR DESCRIPTION
**Pull request by dwkaminsky**
## Description
Updated FlatButton, RaisedButton, and IconButton to be a single Button component, with props to determine which button type it is.

captured (1)

## Motivation and Context
Closes #406
Closes #407
Closes #409

### CODE REVIEW

File: src/lib/components/Button.react.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+   * which will then be displayed. (In our example,
+   * we are nesting an `<input type="file" />` and a `span`
+   * that acts as our label to be displayed.) This only
+   * applies to flat and disableShadow buttons.
+   */
+  children?: Node,
+  /** Mapping from MUI Element name to the intended classname for that Element
+   * See https://material-ui.com/api/button/#css
+   * and https://material-ui.com/api/icon-button/#css*/
+  classes?: Object,
```
Comment (mjclawar) [inline src/lib/components/Button.react.js:21]: Can replace `Object` with
```js
{
  root?: string,
  ...
}
```

See https://github.com/StratoDem/sd-material-ui/pull/427/files#r471462161

Comment (dwkaminsky) [reply src/lib/components/Button.react.js:21]: Got it, I'll change this for all of the PR's. How does the style look wrong?

Comment (mjclawar) [reply src/lib/components/Button.react.js:21]: image

The raised/flat button labels are not centered vertically

Comment (dwkaminsky) [reply src/lib/components/Button.react.js:21]: I think it's just cut off, here's a better view
image

Comment (mjclawar) [reply src/lib/components/Button.react.js:21]: That's a better view of the problem. The button label is not vertically centered within the button, even though it should be.

Comment (dwkaminsky) [reply src/lib/components/Button.react.js:21]: image
Turns out the problem was just the margin of the children...

**Pull request merged**